Userland/Libraries/LibGUI/Window.cpp

```diff
@@ -915,6 +915,14 @@ bool Window::is_maximized() const
     return WindowServerConnection::the().is_maximized(m_window_id);
 }
 
+void Window::set_maximized(bool maximized)
+{
+    if (is_maximized() == maximized)
+        return;
```

Comment:
This will requires an extra IPC round-trip to the WindowServer, which doesn't seem worth it here.

Reply:
Would it be better to store the state in a member variable like m_frameless, m_fullscreen etc, or just remove the check?

Reply:
Just remove the check and let WindowServer be the only authority on window state :)